docs/docs/tutorials/agentic-ai.md

```diff
@@ -0,0 +1,882 @@
+
+### Conditional workflow
+
+Another frequent need is to invoke a certain agent only if a specific condition is satisfied. For example, it could be useful to categorize a user request before processing it, so that the processing can be done by different agents depending on the category of the request. This can be achieved by using the following `CategoryRouter`
+
+```java
+public interface CategoryRouter {
+
+    @UserMessage("""
+        Analyze the following user request and categorize it as 'legal', 'medical' or 'technical'.
+        In case the request doesn't belong to any of those categories categorize it as 'unknown'.
+        Reply with only one of those words and nothing else.
+        The user request is: '{{request}}'.
+        """)
+    @Agent("Categorizes a user request")
+    RequestCategory classify(@V("request") String request);
+}
+```
+
+that returns a `RequestCategory` enum value.
+
+```java
+public enum RequestCategory {
+    LEGAL, MEDICAL, TECHNICAL, UNKNOWN
+}
+```
+
+In this way, having defined a `MedicalExpert` agent like:
+
+```java
+public interface MedicalExpert {
+
+    @UserMessage("""
+        You are a medical expert.
+        Analyze the following user request under a medical point of view and provide the best possible answer.
+        The user request is {{request}}.
+        """)
+    @Agent("A medical expert")
+    String medical(@V("request") String request);
+}
+```
+and similar `LegalExpert` and `TechnicalExpert` agents, it is possible to create a `ExpertRouterAgent`
+
+```java
+public interface ExpertRouterAgent {
+
+    @Agent
+    String ask(@V("request") String request);
+}
+```
+
+implementing a conditional workflow that invokes the appropriate agent based on the category of the user request.
+
+```java
+CategoryRouter routerAgent = AgentServices
+        .agentBuilder(CategoryRouter.class)
+        .chatModel(BASE_MODEL)
+        .outputName("category")
+        .build();
+
+MedicalExpert medicalExpert = AgentServices
+        .agentBuilder(MedicalExpert.class)
+        .chatModel(BASE_MODEL)
+        .outputName("response")
+        .build();
+LegalExpert legalExpert = AgentServices
+        .agentBuilder(LegalExpert.class)
+        .chatModel(BASE_MODEL)
+        .outputName("response")
+        .build();
+TechnicalExpert technicalExpert = AgentServices
+        .agentBuilder(TechnicalExpert.class)
+        .chatModel(BASE_MODEL)
+        .outputName("response")
+        .build();
+
+UntypedAgent expertsAgent = AgentServices.conditionalBuilder()
+        .subAgents( cognisphere -> cognisphere.readState("category", RequestCategory.UNKNOWN) == RequestCategory.MEDICAL, medicalExpert)
+        .subAgents( cognisphere -> cognisphere.readState("category", RequestCategory.UNKNOWN) == RequestCategory.LEGAL, legalExpert)
+        .subAgents( cognisphere -> cognisphere.readState("category", RequestCategory.UNKNOWN) == RequestCategory.TECHNICAL, technicalExpert)
```

Comment:
So if multiple subagents pass the condition, then all will be invoked? Sequentially? If sequentially, is the order somehow defined / predictable?